pkg/backends/otlp/group.go

```diff
@@ -0,0 +1,34 @@
+package otlp
+
+import (
+	"golang.org/x/exp/maps"
+
+	"github.com/atlassian/gostatsd/pkg/backends/otlp/internal/data"
+)
+
+type Group map[string]data.ResourceMetrics
+
+func NewGroup() Group {
+	return make(Group)
+}
+
+func (g *Group) Values() []data.ResourceMetrics {
+	return maps.Values(*g)
+}
+
+func (g *Group) Insert(is data.InstrumentationScope, resources data.Map, metrics ...data.Metric) {
+	key := resources.Hash()
+
+	entry, exist := (*g)[key]
+	if !exist {
+		entry = data.NewResourceMetrics(
+			data.NewResource(
+				data.WithResourceMap(resources),
+			),
+		)
+	}
+	for _, m := range metrics {
+		entry.AppendMetric(is, m)
+	}
+	(*g)[key] = entry
```

Comment:
Do we need a dereference here?

Reply:
Yeah, I wanted to ensure we are referencing the same values and you can't index a pointer to a map so you need to dereference it.